### PR TITLE
Add font name getter

### PR DESCRIPTION
**Pull request by novialriptide**
Fixes: https://github.com/pygame-community/pygame-ce/issues/1969

I dunno if we want `get_name()`, it seems "unpythonic" in my opinion. Should we discuss deprecating the `get_x()` in a future pr? We could just remove them from the docs to act like they never existed.

This article/post could be a good reference for deprecating getter/setter methods.
https://discuss.python.org/t/use-instance-attributes-or-class-properties-inside-the-class-definition/13290/3

### CODE REVIEW

**Comment by Matiiss** [comment]
Sure, getters/setters that are a `get_` and a complementary `set_` method, could be replaced w/ property getters/setters, however, just getters might as well stay as `get_` methods, since they are not to be changed as in this case with this name, it's read-only, but read-only attributes can cause confusion.

File: src_c/doc/font_doc.h [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #define DOC_PYGAMEFONTSYSFONT "SysFont(name, size, bold=False, italic=False) -> Font\ncreate a Font object from the system fonts"
 #define DOC_PYGAMEFONTFONT "Font(filename, size) -> Font\nFont(pathlib.Path, size) -> Font\nFont(object, size) -> Font\ncreate a new Font object from a file"
 #define DOC_FONTBOLD "bold -> bool\nGets or sets whether the font should be rendered in (faked) bold."
+#define DOC_FONTNAME "name -> str\nGets the font's name."
```
Comment (MyreMylar) [inline src_c/doc/font_doc.h:13]:
```suggestion
#define DOC_PYGAMEFONTNAME "name -> str\nGets the font's name."
```
Annoyingly the freetype module already stole the `DOC_FONTNAME` macro name

File: src_c/font.c [hunk 2]
```diff
@@ -671,6 +680,7 @@ font_set_script(PyObject *self, PyObject *arg)
  * Getters and setters for the pgFontObject.
  */
 static PyGetSetDef font_getsets[] = {
+    {"name", (getter)font_getter_name, NULL, DOC_FONTNAME, NULL},
```
Comment (MyreMylar) [inline src_c/font.c:683]:
```suggestion
    {"name", (getter)font_getter_name, NULL, DOC_PYGAMEFONTNAME, NULL},
```

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

File: src_c/doc/font_doc.h (hunk 1 above)
Comment (Starbuck5) [reply src_c/doc/font_doc.h:13]: This file is autogenerated, so I'm not sure how we'd get around that.

Comment (Starbuck5) [reply src_c/doc/font_doc.h:13]: Figured it out: you need to add it to the `#undef` lines in static.c

**Review by Starbuck5 — APPROVED** [review]
This looks cool, thanks!

File: docs/reST/ref/font.rst [hunk 3]
```diff
@@ -198,6 +198,17 @@ solves no longer exists, it will likely be removed in the future.
 
       .. ## Font.bold ##
 
+   .. attribute:: name
+
+      | :sl:`Gets the font's name.`
+      | :sg:`name -> str`
+
+      Returns the font's name.
```
Comment (Starbuck5) [inline docs/reST/ref/font.rst:206]: I already approved, but this is a small fix if you'd like.

Looking at the pygame.freetype docs https://pyga.me/docs/ref/freetype.html#pygame.freetype.Font.name, they mark their version "Read only" in the content. That would be a good thing to add here.

They also call it the "proper font name," which imo has a nice ring to it.

File: test/font_test.py [hunk 4]
```diff
@@ -548,6 +548,10 @@ def test_size(self):
 
         self.assertNotEqual(size, bsize)
 
+    def test_font_name(self):
+        f = pygame_font.Font(None, 20)
+        self.assertEqual(f.name, "FreeSans")
+
```
Comment (ankith26) [inline test/font_test.py:554]: Another suggestion, for improving tests. We should test that trying to set this attr fails with the expected error

**Comment by novialriptide** [comment]
Once all tests have passed, I'll merge this.

**Pull request merged**